backend/server.js

```diff
@@ -62,6 +62,45 @@ hapi.ext({
 sbp('okTurtles.data/set', SERVER_INSTANCE, hapi)
 
 sbp('sbp/selectors/register', {
+  'backend/server/persistState': async function (deserializedHEAD: Object, entry: string) {
+    const contractID = deserializedHEAD.contractID
+    const cheloniaState = sbp('chelonia/private/state')
+    // If the contract has been removed or the height hasn't been updated,
+    // there's nothing to persist
+    if (!cheloniaState.contracts[contractID] || cheloniaState.contracts[contractID].height < deserializedHEAD.head.height) {
+      return
+    }
+    // If the current HEAD is not what we expect, don't save (the state could
+    // have been updated by a later message). This ensures that we save the
+    // latest state and also reduces the number of write operations
+    if (cheloniaState.contracts[contractID].HEAD === deserializedHEAD.hash) {
+      // Extract the parts of the state relevant to this contract
+      const state = {
+        state: cheloniaState[contractID],
+        contractState: cheloniaState.contracts[contractID]
```

Comment:
Is `contractState` the right name for this attribute? It sounds like ... the contract state. But this isn't the contract state, it's internal Chelonia metadata about the state isn't it?

Reply:
I wasn't sure what an appropriate name would be, since they are collectively part of the chelonia state.